src/nni_manager/training_service/dlts/dltsTrainingService.ts

```diff
@@ -0,0 +1,473 @@
+        }
+    }
+
+    public async getClusterMetadata(key: string): Promise<string> {
+        return '';
+    }
+
+    public async cleanUp(): Promise<void> {
+        this.log.info('Stopping DLTS training service...');
+        this.stopping = true;
+
+        const deferred: Deferred<void> = new Deferred<void>();
+        const restServer: DLTSJobRestServer = component.get(DLTSJobRestServer);
+        try {
+            await restServer.stop();
+            deferred.resolve();
+            this.log.info('DLTS Training service rest server stopped successfully.');
+        } catch (error) {
+            // tslint:disable-next-line: no-unsafe-any
+            this.log.error(`DLTS Training service rest server stopped failed, error: ${error.message}`);
+            deferred.reject(error);
+        }
+
+        return deferred.promise;
+    }
+
+    private async submitTrialJobToDLTS(trialJobId: string): Promise<boolean> {
+        const trialJobDetail: DLTSTrialJobDetail | undefined = this.trialJobsMap.get(trialJobId);
+
+        if (trialJobDetail === undefined) {
+            throw new Error(`Failed to find DLTSTrialJobDetail for job ${trialJobId}`);
+        }
+
+        if (this.dltsClusterConfig === undefined) {
+            throw new Error('DLTS Cluster config is not initialized');
+        }
+        if (this.dltsTrialConfig === undefined) {
+            throw new Error('trial config is not initialized');
+        }
+
+        if (this.dltsRestServerPort === undefined) {
+            const restServer: DLTSJobRestServer = component.get(DLTSJobRestServer);
+            this.dltsRestServerPort = restServer.clusterRestServerPort;
+        }
+
+        // Step 1. Prepare DLTS job configuration
+
+        const trialLocalFolder = path.join(getExperimentRootDir(), 'trials-local', trialJobId);
+        //create tmp trial working folder locally.
+        await execMkdir(trialLocalFolder);
+
+        const runScriptContent: string = CONTAINER_INSTALL_NNI_SHELL_FORMAT;
+        // Write NNI installation file to local tmp files
+        await fs.promises.writeFile(path.join(trialLocalFolder, 'install_nni.sh'), runScriptContent, { encoding: 'utf8' });
+
+        // Write file content ( parameter.cfg ) to local tmp folders
+        if (trialJobDetail.form !== undefined) {
+            await fs.promises.writeFile(
+                path.join(trialLocalFolder, generateParamFileName(trialJobDetail.form.hyperParameters)),
+                trialJobDetail.form.hyperParameters.value, { encoding: 'utf8' }
+            );
+        }
+        // tslint:disable-next-line: strict-boolean-expressions
+        const nniManagerIp: string = this.nniManagerIpConfig ? this.nniManagerIpConfig.nniManagerIp : getIPV4Address();
+        const version: string = this.versionCheck ? await getVersion() : '';
+        const nniDLTSTrialCommand: string = String.Format(
+            DLTS_TRIAL_COMMAND_FORMAT,
+            trialLocalFolder,
```

Comment:
Why `NNI_SYS_DIR` is a local folder?

Reply:
Both manager job and trial job will share the same NFS mounting, they share the same directory structure.

Reply:
Is the path which `const trialLocalFolder = path.join(getExperimentRootDir(), 'trials-local', trialJobId)` specified a NFS sharing folder? Does users need to know this folder path and execute mount command before they start NNI job?

Reply:
TLDR; we will provide a NNI manager job template in DLTS to handle these mounting for user.

In my understanding, there will be two directories in use in NNI:
- Config directory, since users have to put config / codes to samba directory first, which will synced to NFS by DLTS.
- `tempdir` to generate transpiled trial code, we temporarily set `TMPDIR=/work/tmp` to make NNI generate transpiled code to mounted directories, it's better to make it configurable.